tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/PutRecordsIntegration.cs

```diff
@@ -0,0 +1,58 @@
+// <copyright file="PutRecordsIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Kinesis
+{
+    /// <summary>
+    /// AWSSDK.Kinesis PutRecords CallTarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.Kinesis",
+        TypeName = "Amazon.Kinesis.AmazonKinesisClient",
+        MethodName = "PutRecords",
+        ReturnTypeName = "Amazon.Kinesis.Model.PutRecordsResponse",
+        ParameterTypeNames = new[] { "Amazon.Kinesis.Model.PutRecordsRequest" },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsKinesisCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class PutRecordsIntegration
+    {
+        private const string Operation = "PutRecords";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TPutRecordsRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the Kinesis operation</param>
+        /// <returns>CallTarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TPutRecordsRequest>(TTarget instance, TPutRecordsRequest request)
```

Comment:
I also don't see an `OnMethodEnd` in this integration, is there something that will close the new scope?

Reply:
On it!

Reply:
I've added the `OnMethodEnd` in a87def01baf7fd2c7dfd02c43213b9203138ce0d , since this required running on .NET Framework to test